extensions/cornerstone/src/Viewport/OHIFCornerstoneViewport.tsx

```diff
@@ -28,428 +29,401 @@ const STACK = 'stack';
-    setImageScrollBarHeight();
+        cornerstoneViewportService.storePresentation({ viewportId });
 
-    return () => {
-      const viewportInfo = cornerstoneViewportService.getViewportInfo(viewportId);
+        // This should be done after the store presentation since synchronizers
+        // will get cleaned up and they need the viewportInfo to be present
+        cleanUpServices(viewportInfo);
 
-      if (!viewportInfo) {
-        return;
-      }
+        if (onElementDisabled && typeof onElementDisabled === 'function') {
+          onElementDisabled(viewportInfo);
+        }
 
-      cornerstoneViewportService.storePresentation({ viewportId });
+        cornerstoneViewportService.disableElement(viewportId);
 
-      // This should be done after the store presentation since synchronizers
-      // will get cleaned up and they need the viewportInfo to be present
-      cleanUpServices(viewportInfo);
+        eventTarget.removeEventListener(Enums.Events.ELEMENT_ENABLED, elementEnabledHandler);
+      };
+    }, []);
+
+    // subscribe to displaySet metadata invalidation (updates)
+    // Currently, if the metadata changes we need to re-render the display set
+    // for it to take effect in the viewport. As we deal with scaling in the loading,
+    // we need to remove the old volume from the cache, and let the
+    // viewport to re-add it which will use the new metadata. Otherwise, the
+    // viewport will use the cached volume and the new metadata will not be used.
+    // Note: this approach does not actually end of sending network requests
+    // and it uses the network cache
+    useEffect(() => {
+      const { unsubscribe } = displaySetService.subscribe(
+        displaySetService.EVENTS.DISPLAY_SET_SERIES_METADATA_INVALIDATED,
+        async ({
+          displaySetInstanceUID: invalidatedDisplaySetInstanceUID,
+          invalidateData,
+        }: Types.DisplaySetSeriesMetadataInvalidatedEvent) => {
+          if (!invalidateData) {
+            return;
+          }
+
+          const viewportInfo = cornerstoneViewportService.getViewportInfo(viewportId);
+
+          if (viewportInfo.hasDisplaySet(invalidatedDisplaySetInstanceUID)) {
+            const viewportData = viewportInfo.getViewportData();
+            const newViewportData = await cornerstoneCacheService.invalidateViewportData(
+              viewportData,
+              invalidatedDisplaySetInstanceUID,
+              dataSource,
+              displaySetService
+            );
+
+            const keepCamera = true;
+            cornerstoneViewportService.updateViewport(viewportId, newViewportData, keepCamera);
+          }
+        }
+      );
+      return () => {
+        unsubscribe();
+      };
+    }, [viewportId]);
 
-      if (onElementDisabled) {
-        onElementDisabled(viewportInfo);
+    useEffect(() => {
+      // handle the default viewportType to be stack
+      if (!viewportOptions.viewportType) {
+        viewportOptions.viewportType = STACK;
```

Comment:
Grab the default type from the display set first, and then default to stack after that.